libs/pruviloj/Pruviloj.idr

```diff
@@ -0,0 +1,115 @@
+module Pruviloj
+
+import Language.Reflection.Elab
+import Language.Reflection.Utils
+
+data Infer : Type where
+  MkInfer : (a : Type) -> a -> Infer
+
+||| Run something for effects, throwing away the return value
+ignore : Functor f => f a -> f ()
+ignore x = map (const ()) x
+
+||| Do nothing
+skip : Applicative f => f ()
+skip = pure ()
+
+||| Attempt to apply a tactic. If it fails, do nothing.
+try : Elab a -> Elab ()
+try tac = tac *> pure () <|> pure ()
```

Comment:
Perhaps `ignore tac` instead of `tac *> pure ()`? Also the function is generic to any `Alternative`.